### PR TITLE
New fetch approach

### PR DESCRIPTION
**Pull request by richfitz**
This PR separates fetching remote data from doing the final curation. It also uses R's DESCRIPTION parsing to do a more reliable job of picking out dependencies (I looked but did not find a good implementation for this in python, despite it being a format that is used in Debian).

Currently it does not generate the exact metadata needed for the existing implementation, but is close

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,7 +33,19 @@ follows, requiring `python3` (3.7 or later):
...
+```bash
+./scripts/run_fetch
+```
+
+1. Parse the DESCRIPTION files, which requires an R installation
+
+```bash
+./scripts/run_parse_description
+```
+
```
Comment (hillalex) [inline README.md:52]:
```suggestion
1. Generate a json file containing metadata in a format that we will use

```bash
./scripts/run_generate_json
```

```

File: scripts/parse.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import csv
+import json
+import os
+
+from pathlib import Path
+
+
+def read_packages(path, config):
+    ret = {}
+    for p in Path('data').rglob('metadata.json'):
```
Comment (hillalex) [inline scripts/parse.py]:
```suggestion
    for p in Path(path).rglob('metadata.json'):
```

File: scripts/run_generate_json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python3
+import csv
+import json
+
+from parse import *
+from config import Config
+
+root = "."
```
Comment (hillalex) [inline scripts/run_generate_json]: can we wrap this in
```
import os

if __name__ == "__main__":
    root = os.path.dirname(os.path.realpath(__file__))
```

File: scripts/run_parse_description [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      error = function(e) {
+        message(sprintf("Parse failed for %s: %s", p, e$message))
+      })
+  }
+}
+
+main <- function() {
+  options(warn = 1) # immediate warnings
+  ensure_packages(c("desc", "jsonlite"))
+  process_path("data")
```
Comment (hillalex) [inline scripts/run_parse_description]:
```suggestion
  process_path("scripts/data")
```

Needed if running from top level dir as suggested by README

**Review by hillalex — CHANGES_REQUESTED** [review]
This all looks good. Some minor changes I needed to get this running as desired (and then I can make a small tweak to the graph generation to get that working)

**Comment by hillalex** [comment]
Additonally, looks like for Hugo templating json files should be written to the `data` directory and it will be preferable to have an array rather than a dictionary.

So `data/repos.json` where `repos.json` contains an array `[ { "repo": "reside-ic/rb.test.json", ... },... ]`

File: scripts/run_parse_description (hunk 4 above)
Comment (richfitz) [reply scripts/run_parse_description]: done via path discovery, which is much uglier than the python version

**Review by hillalex — APPROVED** [review]
perfect, thanks!

**Pull request merged**